### PR TITLE
Issue 15639: Initial messages for Ldap Kerberos

### LINKED ISSUE #15639: KRB5: Update and reuse Kerberos messages
We can reuse several of the messages from WebSphere Application Server. Update and reuse as appropriate.

 #12794

```
<Message ID="CWWIM5120E" prefix="yes">
		<MsgText pgmKey="KRB5_LOGIN_FAILED_CACHE">
		    Kerberos login failed using Kerberos principal {0} and Kerberos credential cache (ccache) {1}.
		</MsgText>
		<Explanation>
			<p>Either the specified Kerberos principal is invalid, or the Kerberos credential cache (ccache) is invalid or expired.</p>
		</Explanation>
		<ProgrammerResponse>
			<p>Ensure a valid Kerberos principal is specified and the Kerberos credential cache (ccache) has not expired.</p>

<Message ID="CWWIM5121E" prefix="yes">
		<MsgText pgmKey="KRB5_LOGIN_FAILED_KEYTAB">
		    Kerberos login failed using Kerberos principal {0} and Kerberos keytab {1}.
		</MsgText>
		<Explanation>
			<p>Either the specified Kerberos principal is invalid, or the Kerberos keytab is invalid.</p>
		</Explanation>
		<ProgrammerResponse>
			<p>Ensure a valid Kerberos principal is specified and a valid Kerberos keytab containing the Kerberos principal.</p>
		
<Message ID="CWWIM5122E" prefix="yes">
		<MsgText pgmKey="KRB5_LOGIN_FAILED_DEFAULT_CACHE">
		    Kerberos login failed using Kerberos principal {0} and the default Kerberos credential cache (ccache).
		</MsgText>
		<Explanation>
			<p>Either the specified Kerberos principal is invalid, or the default Kerberos credential cache (ccache) is invalid or expired.</p>
		</Explanation>
		<ProgrammerResponse>
			<p>Ensure a valid Kerberos principal is specified and the default Kerberos credential cache (ccache) has not expired.</p>

<Message ID="CWWIM5123E" prefix="yes">
		<MsgText pgmKey="KRB5_LOGIN_FAILED_DEFAULT_KEYTAB">
		    Kerberos login failed using Kerberos principal {0} and the default Kerberos keytab.
		</MsgText>
		<Explanation>
			<p>Either the specified Kerberos principal is invalid, or the default Kerberos keytab is invalid.</p>
		</Explanation>
		<ProgrammerResponse>
			<p>Ensure a valid Kerberos principal is specified and the default Kerberos keytab is valid.</p>
		</ProgrammerResponse>

<Message ID="CWWIM5124I" prefix="yes">
		<MsgText pgmKey="KRB5_TICKETCACHE_USED">
		    LDAPRegistry {0} configured with Kerberos credential cache (ccache) filename {1} and keytab filename {2}, using Kerberos credential cache (ccache) for Kerberos bind authentication to LDAP server.
		</MsgText>
		<Explanation>
			<p>In order to use keytab, the keytab filename must be specified and the Kerberos credential cache (ccache) filename must not be specified.</p>
		</Explanation>
		<ProgrammerResponse>
			<p>No action is required.</p>
		

<Message ID="CWWIM5125E" prefix="yes">
		<MsgText pgmKey="FAILED_GET_KRB5_SECURITY_CONFIG">
		    The {0} LDAPRegistry did not load the Kerberos keytab and configuration file from the security configuration. The error that was received is {1}.
		</MsgText>
		<Explanation>
			<p>The Kerberos keytab and configuration files are saved in a single global location in the security configuration. To get or set the LDAP Kerberos configuration, the keytab and configuration file names are loaded from the security configuration.</p>
		</Explanation>
		<ProgrammerResponse>
			<p>Review the provided exception.</p>

<Message ID="CWWIM5126E" prefix="yes">
		<MsgText pgmKey="FAILED_SET_KRB5_SECURITY_CONFIG">
		    The {0} LDAPRegistry did not load the Kerberos keytab and configuration file from the security configuration. The error that was received is {1}.
		</MsgText>
		<Explanation>
			<p>The Kerberos keytab and configuration files are saved in a single global location in the security configuration. To set the LDAP Kerberos configuration, the keytab and configuration file names are saved in the security configuration.</p>
		</Explanation>
		<ProgrammerResponse>
			<p>Review the provided exception.</p>

<Message ID="CWWIM5128E" prefix="yes">
		<MsgText pgmKey="INVALID_KRB5_PRINCIPAL">
		    The {0} Kerberos principal name is incorrectly formatted, or the realm name is missing, or a default realm name cannot be found.
		</MsgText>
		<Explanation>
			<p>The Kerberos principal name cannot be null. The principle name must either include the realm name or a default realm name must be defined in Kerberos configuration file.</p>
		</Explanation>
		<ProgrammerResponse>
			<p>Correct the principal name or add a default realm name.</p

<Message ID="CWWIM5129E" prefix="yes">
		<MsgText pgmKey="CANNOT_READ_KRB5_FILE">
		    LDAPRegistry {0} could not read the Kerberos file {1}.
		</MsgText>
		<Explanation>
			<p>The Kerberos file cannot be opened because either the file permissions are incorrect or the file does not exist.</p>
		</Explanation>
		<ProgrammerResponse>
			<p>Verify that the file location is correct and the server has read file permissions.</p>
```

### PR DESCRIPTION
**Pull request by kristip17**
Fixes #15639

### CODE REVIEW

**Comment by kristip17** [comment]
Ready for message review. FYI @ManasiGandhi -- these are similar to ones that you recently did for WAS.

**Comment by ManasiGandhi** [comment]
Hi Kristi,
 
I think Jeff is reviewing the messages currently.
 
Best,
Manasi
 
----- Original message -----From: Kristi Peterson <notifications@github.com>To: OpenLiberty/open-liberty <open-liberty@noreply.github.com>Cc: Manasi Gandhi <manasi.gandhi@ibm.com>, Mention <mention@noreply.github.com>Subject: [EXTERNAL] Re: [OpenLiberty/open-liberty] Issue 15639: Initial messages for Ldap Kerberos (#15659)Date: Mon, Jan 25, 2021 2:38 PM 
 
Ready for message review. FYI @ManasiGandhi -- these are similar to ones that you recently did for WAS.

File: dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops [hunk 1]
```diff
@@ -270,4 +270,36 @@ CANNOT_WRITE_TO_READ_ONLY_REPOSITORY.useraction=Ensure that a write operation is
...
+KRB5_LOGIN_FAILED_CACHE.explanation=Either the specified Kerberos principal is invalid, or the Kerberos credential cache (ccache) is invalid or expired.
+KRB5_LOGIN_FAILED_CACHE.useraction=Ensure a valid Kerberos principal is specified and the Kerberos credential cache (ccache) has not expired.
+
+KRB5_LOGIN_FAILED_KEYTAB=CWIML4554E: Kerberos login failed using Kerberos principal {0} and Kerberos keytab {1}.
+KRB5_LOGIN_FAILED_KEYTAB.explanation=Either the specified Kerberos principal is invalid, or the Kerberos keytab is invalid.
+KRB5_LOGIN_FAILED_KEYTAB.useraction=Ensure a valid Kerberos principal is specified and a valid Kerberos keytab containing the Kerberos principal.
+		
+KRB5_LOGIN_FAILED_DEFAULT_CACHE=CWIML4555E: Kerberos login failed using Kerberos principal {0} and the default Kerberos credential cache (ccache).
+KRB5_LOGIN_FAILED_DEFAULT_CACHE.explanation=Either the specified Kerberos principal is invalid, or the default Kerberos credential cache (ccache) is invalid or expired.
+KRB5_LOGIN_FAILED_DEFAULT_CACHE.useraction=Ensure a valid Kerberos principal is specified and the default Kerberos credential cache (ccache) has not expired.
```
Comment (jantley-ibm) [inline dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops:285]: same comment as above

File: dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops [hunk 2]
```diff
@@ -270,4 +270,36 @@ CANNOT_WRITE_TO_READ_ONLY_REPOSITORY.useraction=Ensure that a write operation is
...
+
+KRB5_LOGIN_FAILED_DEFAULT_KEYTAB=CWIML4556E: Kerberos login failed using Kerberos principal {0} and the default Kerberos keytab.
+KRB5_LOGIN_FAILED_DEFAULT_KEYTAB.explanation=Either the specified Kerberos principal is invalid, or the default Kerberos keytab is invalid.
+KRB5_LOGIN_FAILED_DEFAULT_KEYTAB.useraction=Ensure a valid Kerberos principal is specified and the default Kerberos keytab is valid.
```
Comment (jantley-ibm) [inline dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops:289]: Ensure that a valid Kerberos principal is specified and that the default Kerberos keytab is valid.

File: dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops [hunk 3]
```diff
@@ -270,4 +270,36 @@ CANNOT_WRITE_TO_READ_ONLY_REPOSITORY.useraction=Ensure that a write operation is
...
+
+KRB5_TICKETCACHE_USED=CWIML4557I: The {0} LdapRegistry is configured with a Kerberos ticket cache (ccache) filename {1} and keytab filename {2}. The Kerberos credential cache (ccache) is used for Kerberos bind authentication to LDAP server.
+KRB5_TICKETCACHE_USED.explanation=In order to use the keytab file, the keytab filename must be specified and the Kerberos credential cache (ccache) filename must not be specified.
```
Comment (jantley-ibm) [inline dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops:292]: change
In order to use the keytab file,
to
To use the keytab file,

File: dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops [hunk 4]
```diff
@@ -270,4 +270,36 @@ CANNOT_WRITE_TO_READ_ONLY_REPOSITORY.useraction=Ensure that a write operation is
...
+KRB5_TICKETCACHE_USED.useraction=No action is required. To avoid this message, remove the LdapRegistry ticketCache attribute or the Kerberos keytab attribute.
```
Comment (jantley-ibm) [inline dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops:293]: maybe change:
remove the
to:
remove either the

File: dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops [hunk 5]
```diff
@@ -270,4 +270,36 @@ CANNOT_WRITE_TO_READ_ONLY_REPOSITORY.useraction=Ensure that a write operation is
...
+
+INVALID_KRB5_PRINCIPAL=CWIML4558E: The {0} Kerberos principal name is incorrectly formatted, or the realm name is missing, or a default realm name cannot be found.
+INVALID_KRB5_PRINCIPAL.explanation=The Kerberos principal name cannot be null. The principal name must either include the realm name or a default realm name must be defined in Kerberos configuration file.
```
Comment (jantley-ibm) [inline dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops:296]: Second sentence is confusing; suggested change to fix:
The principal name must either include the realm name
to
Either the principal name must include the realm name,

File: dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops [hunk 6]
```diff
@@ -270,4 +270,36 @@ CANNOT_WRITE_TO_READ_ONLY_REPOSITORY.useraction=Ensure that a write operation is
...
+INVALID_KRB5_PRINCIPAL.useraction=Correct the principal name or add a default realm name.
+
+CANNOT_READ_KRB5_FILE=CWIML4559E:The {0} LdapRegistry could not read the Kerberos file at {1}.
```
Comment (jantley-ibm) [inline dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops:299]: change
could not read
to
cannot read

File: dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops [hunk 7]
```diff
@@ -270,4 +270,36 @@ CANNOT_WRITE_TO_READ_ONLY_REPOSITORY.useraction=Ensure that a write operation is
...
+CANNOT_READ_KRB5_FILE.explanation=The Kerberos file cannot be opened because either the file permissions are incorrect or the file does not exist.
```
Comment (jantley-ibm) [inline dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops:300]: Suggested rewrite:
The Kerberos file cannot be opened. Either the file permissions are incorrect, or the file does not exist.

File: dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops [hunk 8]
```diff
@@ -270,4 +270,36 @@ CANNOT_WRITE_TO_READ_ONLY_REPOSITORY.useraction=Ensure that a write operation is
...
+CANNOT_READ_KRB5_FILE.useraction=Verify that the file location is correct and the server has read file permissions.
```
Comment (jantley-ibm) [inline dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops:301]: change
and the server
to
and that the server

**Review by jantley-ibm — COMMENTED** [review]
Suggested message rewrites attached. Let me know if you have questions or concerns. Will add ID reviewed tag when we agree on changes to the PR. Thanks --

**Comment by skasund** [comment]
L2 message review completed

**Comment by kristip17** [comment]
Thanks @jantley-ibm -- I believe I have made all of the requested updates.

**Comment by kristip17** [comment]
#build

**Comment by kristip17** [comment]
#libbybot

**Review by jvanhill — APPROVED** [review]

**Pull request merged**